validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/MutableLocalValidatorSource.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.loader;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.List;
+import org.apache.commons.lang3.NotImplementedException;
+import tech.pegasys.signers.bls.keystore.model.KeyStoreData;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.core.signatures.DeletableSigner;
+import tech.pegasys.teku.core.signatures.Signer;
+import tech.pegasys.teku.core.signatures.SlashingProtector;
+
+public class MutableLocalValidatorSource extends SlashingProtectedValidatorSource {
+
+  public MutableLocalValidatorSource(
+      final ValidatorSource delegate, final SlashingProtector slashingProtector) {
```

Comment:
I suspect we should explicitly require a `LocalValidatorSource` for the delegate here since we're requiring it to be able to load keystrokes from the file system that we'll wind up writing to.

In fact, this is probably the time for a static create method that takes only the `SlashingProtector` and storage directory and creates the `LocalValidatorSource` internally since it should also read from the same directory we're writing to.

Not against doing that in a follow up PR though when we actually come to writing the keystores.